p2p/host/autorelay/autorelay.go

```diff
@@ -135,6 +138,75 @@ func (ar *AutoRelay) background(ctx context.Context) {
 	}
 }
 
+func (ar *AutoRelay) refresh(ctx context.Context) {
+	ticker := time.NewTicker(time.Minute)
+	defer ticker.Stop()
+
+	for {
+		select {
+		case <-ticker.C:
+			var toRefresh []peer.ID
+
+			ar.mx.Lock()
+			if ar.status == network.ReachabilityPublic {
+				// we are public, forget about the relays, unprotect peers
+				for p := range ar.relays {
+					ar.host.ConnManager().Unprotect(p, "autorelay")
+					delete(ar.relays, p)
+				}
+
+				ar.mx.Unlock()
+				continue
+			}
+
+			// find reservations about to expire
+			now := time.Now()
+			for p, rsvp := range ar.relays {
+				if rsvp == nil {
+					continue
+				}
+
+				if now.Add(time.Minute).Before(rsvp.Expiration) {
```

Comment:
If we're running this loop once per minute, we should probably refresh 2 minutes before the expiration, to be on the safe side. (Best to introduce a const for that logic).

Reply:
good point, will do.

Reply:
fixed.